config/yaml_test.go

```diff
@@ -56,6 +56,39 @@ func TestYamlSimple(t *testing.T) {
 	assert.Equal(t, ":28941", c.AsString())
 }
 
+func TestYAMLEnvInterpolation(t *testing.T) {
+	t.Parallel()
+	defer env.Override(t, "OWNER_EMAIL", "hello@there.yasss")()
+
+	cfg := []byte(`
+name: some name here
+owner: ${OWNER_EMAIL}
+module:
+  fake:
+    number: ${FAKE_NUMBER:321}`)
```

Comment:
I think empty default is legit, i.e. `fullTelephone: 1-800-LOLZ${OPTIONAL_EXTENSION:}`

Should result in `1-800-LOLZ` if `OPTIONAL_EXTENSION` is not present in env. I am happy to discuss if we want a special syntax for that, but right now doing  `fullTelephone: 1-800-LOLZ${OPTIONAL_EXTENSION}` will panic, since there is no fallback at all